gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/services/impl/AccountServiceImpl.java

```diff
@@ -185,6 +184,25 @@ public Single<Credential> getWebAuthnCredential(String id) {
                 });
     }
 
+    @Override
+    public Single<List<ScopeApproval>> getConsentList(User user, Client client) {
+        Flowable<ScopeApproval> source = (client != null) ?
+                scopeApprovalService.findByDomainAndUserAndClient(domain.getId(),  user.getId(), client.getClientId()) :
+                scopeApprovalService.findByDomainAndUser(domain.getId(), client.getId());
```

Comment:
```suggestion
                scopeApprovalService.findByDomainAndUser(domain.getId(), user.getId());
```

Also can user be null ? 
Maybe it's safer to go with

```suggestion

if (client != null){
  return scopeApprovalService.findByDomainAndUserAndClient(domain.getId(),  user.getId(), client.getClientId()).toList()
} else if (user != null){
  return scopeApprovalService.findByDomainAndUser(domain.getId(), user.getId()).toList()
}

return Single.just(List.of());
```